### PR TITLE
fix(number mask): [SE-618] use `toLocaleString` instead of custom implementation

### PR DESCRIPTION
**Pull request by jordanCain**
Tested on web-client and web-admin asserting the values returned from this refactor match what was previously returned

### CODE REVIEW

File: src/numberMask/index.js [hunk 1]
```diff
@@ -1,20 +1,14 @@
...
-
+/**
+ * Given a Number, return a masked version with commas and squash to 2 fractionals
+ * @param {Number} value Number to be masked
+ * @returns String
+ */
 const numberMask = (value) => {
-  const stringValue = String(value);
-  const isNegative = stringValue.charAt(0) === '-';
+  if (value === null || value === 0 || value === '') return 0;
```
Comment (rhamzeh) [inline src/numberMask/index.js:7]: Can we just say

```
if (!value) { ... }
```

I feel if value is false or undefined, we want to also return 0

**Review by rhamzeh — CHANGES_REQUESTED** [review]

File: src/numberMask/index.js [hunk 2]
```diff
@@ -1,20 +1,15 @@
...
+ * Given a Number, return a masked version with commas and squash to 2 fractionals
+ * @param {Number} value Number to be masked
+ * @returns String
+ */
 const numberMask = (value) => {
-  const stringValue = String(value);
-  const isNegative = stringValue.charAt(0) === '-';
+  if (!value) return '';
+  if (value === '-') return '-';
+  const valueAsNumber = parseFloat(value);
```
Comment (jordanCain) [inline src/numberMask/index.js:9]: Problem with parseFlow/parseInt is that if a string is passed in by mistake the parsing will end at the first none numerical character, e.g `9,999` would be parsed as `9`

Comment (jordanCain) [reply src/numberMask/index.js:9]: We could cast the input parameter to a string and then try and do a regex replace of [non numeric, '.', '-'], if we did this we would need the same previous regex used to track for negative numbers I think ```const numberValue = isNegative ? `-${removeNonNumeric(R.tail(stringValue))}` : removeNonNumeric(stringValue);``` due to lack of browser compatibility with negative look behind regex

Comment (jordanCain) [reply src/numberMask/index.js:9]: If we just tried to remove all non numeric values first we would run the risk of modifying a string that gets passed in such as "-1" and returning "1"

File: src/numberMask/index.js [hunk 3]
```diff
@@ -1,20 +1,15 @@
...
-
+/**
+ * Given a Number, return a masked version with commas and squash to 2 fractionals
+ * @param {Number} value Number to be masked
+ * @returns String
+ */
 const numberMask = (value) => {
-  const stringValue = String(value);
-  const isNegative = stringValue.charAt(0) === '-';
+  if (!value) return '';
```
Comment (rhamzeh) [inline src/numberMask/index.js:7]: This PR is almost 💯 

But minor thing, if `value === 0` we need to return `0` not an empty string

**Review by rhamzeh — CHANGES_REQUESTED** [review]

**Review by rhamzeh — APPROVED** [review]

**Pull request merged**